ui/src/pages/AllStories/index.tsx

```diff
@@ -133,10 +158,28 @@ function createData({
     image_url,
     video_url,
     content,
+    tags,
   };
 }
 
 export const AllStories: React.FC = () => {
+  const [anchorEl, setAnchorEl] = React.useState<HTMLButtonElement | null>(
+    null
+  );
+
+  const handleClick = (event: React.MouseEvent<HTMLButtonElement>) => {
+    setAnchorEl(event.currentTarget);
+  };
+
+  const handleClose = () => {
+    setAnchorEl(null);
+  };
+
+  const open = Boolean(anchorEl);
```

Comment:
nit: more descriptive name
```suggestion
  const isFilterOpen = Boolean(anchorEl);
```